python/ray/data/_internal/arrow_block.py

```diff
@@ -60,17 +55,6 @@
 T = TypeVar("T")
 logger = logging.getLogger(__name__)
 
-ARROW_OBJECT_FIXABLE_ERRORS = (
-    pyarrow.lib.ArrowTypeError,
-    pyarrow.lib.ArrowNotImplementedError,
-    pyarrow.lib.ArrowInvalid,
-)
-
-
-def is_object_fixable_error(e: ArrowConversionError) -> bool:
-    """Returns whether this error can be fixed by using an ArrowPythonObjectArray"""
-    return isinstance(e.__cause__, ARROW_OBJECT_FIXABLE_ERRORS)
-
```

Comment:
These are removed, `ArrowPythonObjectType` extension is now applied unconditionally whenever native Arrow fails to serialize batch